test/plugins/fingerprint.pulgin.spec.ts

```diff
@@ -0,0 +1,102 @@
+
+    constructor(private readonly status: string, private readonly id: string, private readonly agentId: string) {}
+
+    externalIds: object;
+    async getIds(filters: VerifyFiltersDto): Promise<Map<string, string>> {
+        return new Map<string, string>(Object.entries(filters.externalIds));
+    }
+
+    async verify(params: any, filters: VerifyFiltersDto): Promise<{}> {
+        return Promise.resolve({
+            data: {
+                status: this.status,
+                did: this.id,
+            },
+        });
+    }
+
+    async verifyFingerprint(params: any, position: number, image: string, agentId: string): Promise<any> {
+        return Promise.resolve({
+            data: {
+                status: this.status,
+                agentId: this.agentId,
+            },
+        });
+    }
+}
+describe('Fingerprint Plugin is valid', () => {
+    //const fingerprintVerifySpec = new VerifyFiltersDto ;
+    it('Plugin is not empty', async () => {
+        const agentId = 'agentId123';
+        const bioAuthService = new MockBioAuthService('not_matched', agentId);
+
+        const id2 = 1000000 + parseInt(now().toString().substr(7,6), 10);
+        const id1 = 'N' + id2;
+        const id1Hash = pepperHash(id1);
+        const mockExternalId = new ExternalId();
+        mockExternalId.agent_id = agentId;
+        mockExternalId.external_id = id1Hash;
+        mockExternalId.external_id_type = 'id_1';
+        const mockExternalIdRepository: any = new MockRepository<ExternalId>([mockExternalId]);
+        
+        const fpPlugin = new FingerprintPlugin(bioAuthService, mockExternalIdRepository);
+
+        const params = {
+            position: 1,
+            image: 'some image'
+        } as VerifyFingerprintImageDto;
+        const filters = {
+            externalIds: {
+                key: 'value'
+            }
+        } as VerifyFiltersDto;
+        (await expect(fpPlugin.verify(params, filters))).rejects.toThrow(ProtocolException);
+
+    });
+    it('If fingerpirnt no match is thrown should throw a Protocol exception', async () => {
+
+        const agentId = 'agentId123';
+        const mockExternalId = new ExternalId();
+        const bioAuthService = new class extends MockBioAuthService {
+            async verifyFingerprint(position: number, image: string, agentIds: string): Promise<any> {
+                throw new ProtocolException(ProtocolErrorCode.FINGERPRINT_NO_MATCH, 'msg');
+            }
+        }('matched', agentId);
+        const mockExternalIdRespository: any = new MockRepository<ExternalId>([mockExternalId]);
+        const fpPlugin = new FingerprintPlugin(bioAuthService, mockExternalIdRespository);
+        const params = {
+            position: 1,
+            image: 'some image'
+        } as VerifyFingerprintImageDto;
+        const filters = {
+            externalIds: {
+                key: 'value'
+            }
+        } as VerifyFiltersDto;
```

Comment:
A lot of these values are common across the two tests you have. You can move them out to a higher scope so that they're not redefined unnecessarily for each test. In particular: `agentId`, `params`, and `filters` are defined identically across the two tests. This is a good place to practice being DRY!